src/pages-and-resources/discussions/data/permissions.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * A wrapper for user permissions.
+ * 
+ * This offers the convenience of checking for permissions,
+ * assuming users can take actions unless explicitly stated.
+ * 
+ * For example, checking whether a user can "change_providers"
+ * can be done simply with
+ * 
+ *      if (permissions.change_providers) { ... }
+ * 
+ * instead of the bare object alternative
+ * 
+ *      if (permissions.change_providers === undefined
+ *          || permissions.change_providers
+ *      ) { ... }
+ */
+export function userPermissions(permissions) {
+    permissions = permissions || {}
+    return new Proxy({...permissions}, {
+        get(target, name, receiver) {
+            const value = Reflect.get(target, name, receiver);
+            return value === undefined || value;
+        }
+    });
+}
```

Comment:
Let's remove this and keep things simple. The reducers can simply normalise the response so that this kind of check is unnecessary. If the UI is using a permission name somewhere, then we are aware of it. 

So, instead you can create an Enum of permissions, and then permission checks can be performed as:
```javascript
user_has_permission(Permissions.CHANGE_PROVIDERS)
```
That function can then do the undefined check if needed. 

I think this kind of complexity only makes sense if we don't already know the shape of the object in advance. Here we already know which permissions exist, so it doesn't really serve the purpose. 

Additionally, won't `value === undefined || value` mean that all permissions default to `true` unless explicitly set `false` by the api response? I think this might be the opposite of what we want. This should probably be `value !== undfefined && value`.

Reply:
The idea was to use this to hold the permission hints returned by the backend (since removed), so it's going away anyway.

Regarding `value === undefined || value`, the rationale was that defaulting to `true` made supporting new permissions in the frontend backwards compatible, i.e., if you deployed the frontend with a new permissions check that the backend did not support yet, nothing would change.
Using this as a concrete example, defaulting to `true` would keep the checkboxes enabled **for all users** (as it is now), while defaulting to `false` would disable them **for all users** (because `userPermissions['change_providers'] === undefined`).

Reply:
I don't really see a use case for the frontend getting a feature based on an API that doesn't exist yet. 
Even in that use case, you can simply list each permission with its default value in the initial data for the reducer, and design the reducer to only update existing data, not replace it. This gives you control over how the permission is used. 

My main point here is that the frontend will always know the list of permissions it is working with. If the backend adds a new permission that the frontend doesn't handle yet, that's OK, that permission will never be queried so the above code won't help. If the frontend adds a permission that the backend doesn't supply yet (an unlikely scenario, except for a very very short while), then it can just initialise it with a sensible default value, and it will never be overridden by a backend value.